src/Json/SchemaLoader.php

```diff
@@ -61,7 +62,12 @@ public function load(string $file): stdClass
         $locatedFile = $this->fileLocator->locate($file);
 
         if (isset($this->schemas[$locatedFile]) === false) {
-            $schema = $this->dereferencer->dereference('file://'.$locatedFile);
+            if (in_array(pathinfo($locatedFile, PATHINFO_EXTENSION), array('yml', 'yaml'))) {
+                $dereference = (object) Yaml::parseFile($locatedFile);
```

Comment:
The Yaml parser can also return an object/stdClass as output:
```suggestion
                $dereference = Yaml::parseFile($locatedFile, Yaml::PARSE_OBJECT_FOR_MAP);
```